src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Database/DatabaseHandler.cs

```diff
@@ -495,11 +495,11 @@ public string Drop(DropDatabaseRequestParams dropParams)
                             var sqlConn = dataContainer.ServerConnection.SqlConnectionObject;
                             SqlConnection.ClearPool(sqlConn);
                         }
+                        smoDatabase.Drop();
```

Comment:
I put this in the other one, but we should consider having these be a transactional operation so that if either of them fails the whole thing is rolled back.